test-lib.sh

```diff
@@ -31,29 +31,38 @@ UNSTABLE_TESTS="${UNSTABLE_TESTS:-""}"
 # Uses: $EXPECTED_EXIT_CODE - expected container exit code
 # Uses: $TESTSUITE_RESULT - overall result of all tests
 function ct_cleanup() {
-  ct_show_resources
-  for cid_file in "$CID_FILE_DIR"/* ; do
-    [ -f "$cid_file" ] || continue
-    local container
-    container=$(cat "$cid_file")
-
-    : "Stopping and removing container $container..."
-    docker stop "$container"
-
-    # Container has not been removed by `docker stop` and still exists
-    if [ "$( docker ps -a -f "id=$container" | wc -l )" -eq 2 ]; then
-      exit_status=$(docker inspect -f '{{.State.ExitCode}}' "$container")
-      if [ "$exit_status" != "$EXPECTED_EXIT_CODE" ]; then
-        : "Dumping logs for $container"
-        docker logs "$container"
+  clean_containers() {
```

Comment:
Is there any reason for subfunction?
What about starting with

```
  if [[ -z ${CID_FILE_DIR:-} ]]; then
    echo "The \$CID_FILE_DIR is not set. Container cleaning is to be skipped."
    return
  fi;
```
and run cleanup in case of CIF_FILE_DIR is set.